scripts/build.py

```diff
@@ -81,30 +99,101 @@ def process_third_party_libs():
+                    if source_path.find(THIRD_PARTY_GENERATED_OUT_DIR) > 0:
+                        continue
+                    if source_path.find(THIRD_PARTY_GENERATED_SRC_DIR) == -1:
+                        continue
+                    target_path = source_path.replace(THIRD_PARTY_GENERATED_SRC_DIR,
+                                                      THIRD_PARTY_GENERATED_OUT_DIR)
+                    process_build(source_path, target_path)
+
+        # Process core/templates/dev/head
+        if any([s in root for s in ['core']]):
+            for root, dirs, files in os.walk(os.path.join(os.getcwd(), root)):
+                if not any([s in root for s in [os.path.join('templates')]]):
+                    continue
+                for directory in dirs:
+                    print 'Processing %s' % os.path.join(root, directory)
+                for filename in files:
+                    source_path = os.path.join(root, filename)
+                    if source_path.find(OUT_DIR) > 0:
+                        continue
+                    if source_path.find(HEAD_DIR) == -1:
+                        continue
+                    target_path = source_path.replace(HEAD_DIR, OUT_DIR)
+                    if filename.endswith('.html'):
+                        process_html(source_path, target_path)
+                    if filename.endswith('.css'):
+                        process_css(source_path, target_path)
+                    if filename.endswith('.js'):
+                        process_js(source_path, target_path)
+
+                # Process core/templates/prod/head/css, copy it to build/[cache_slug]/css
+                if not any([s in root for s in [os.path.join('templates', 'prod', 'head', 'css')]]):
                     continue
-                target_path = source_path.replace(HEAD_DIR, OUT_DIR)
-                if filename.endswith('.html'):
-                    process_html(source_path, target_path)
-                if filename.endswith('.css'):
-                    process_css(source_path, target_path)
-                if filename.endswith('.js'):
-                    process_js(source_path, target_path)
+
+                print 'Processing css directory.'
+
+                for directory in dirs:
+                    print 'Processing %s' % os.path.join(root, directory)
+
+                for filename in files:
+                    source_path = os.path.join(root, filename)
+                    if source_path.find(CSS_OUT_DIR) > 0:
+                        continue
+                    if source_path.find(CSS_SRC_DIR) == -1:
+                        continue
+                    target_path = source_path.replace(CSS_SRC_DIR, CSS_OUT_DIR)
+                    process_build(source_path, target_path)
 
 
 if __name__ == '__main__':
+    # TODO(gvishal): get this cache slug somehow
+    # CACHE_SLUG = utils.get_cache_slug()
+    CACHE_SLUG = 'default'
+    BUILD_DIR = os.path.join('build', CACHE_SLUG)
```

Comment:
It's OK to have these local to this section of the code, and not declare them as global constants. If you've got a function that moves things from one place to the other you can just call that function several times on the constructed dirpaths.